frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/base/link_to_page_widget.dart

```diff
@@ -92,85 +93,167 @@ class LinkToPageMenu extends StatefulWidget {
 }
 
 class _LinkToPageMenuState extends State<LinkToPageMenu> {
+  final _focusNode = FocusNode(debugLabel: 'reference_list_widget');
   EditorStyle get style => widget.editorState.editorStyle;
+  int _selectedIndex = 0;
+  int _totalItems = 0;
+  List<dartz.Tuple2<AppPB, List<ViewPB>>> _availableLayout = [];
+  final Map<int, dartz.Tuple2<AppPB, ViewPB>> _items = {};
+  bool _isLoading = true;
+
+  Future<void> fetchItems() async {
+    final items = await AppBackendService().fetchViews(widget.layoutType);
+
+    setState(() {
+      _availableLayout = items;
+      _isLoading = false;
+    });
+
+    int index = 0;
+    for (final app in _availableLayout) {
+      for (final view in app.value2) {
+        _items.putIfAbsent(index, () => dartz.Tuple2(app.value1, view));
+        index += 1;
+      }
+    }
+
+    setState(() {
+      _totalItems = index;
+    });
+  }
+
+  @override
+  void initState() {
+    fetchItems();
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      _focusNode.requestFocus();
+    });
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _focusNode.dispose();
+    super.dispose();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return Container(
-      color: Colors.transparent,
-      width: 300,
+    return Focus(
+      focusNode: _focusNode,
+      onKey: _onKey,
       child: Container(
-        padding: const EdgeInsets.fromLTRB(10, 6, 10, 6),
-        decoration: BoxDecoration(
-          color: style.selectionMenuBackgroundColor,
-          boxShadow: [
-            BoxShadow(
-              blurRadius: 5,
-              spreadRadius: 1,
-              color: Colors.black.withOpacity(0.1),
-            ),
-          ],
-          borderRadius: BorderRadius.circular(6.0),
+        color: Colors.transparent,
+        width: 300,
+        child: Container(
```

Comment:
A Container with a Container as a Child?

Reply:
Hey! That piece of code existed before, and my bad I didn't check it. I will update that too. Thanks.

Reply:
I assume it made sense when there were padding, depending on what the child's properties are, but without margin/padding, a container with another container as it's child does not make sense.